runelite-client/src/main/java/net/runelite/client/plugins/kourendlibrary/KourendLibraryPanel.java

```diff
@@ -152,6 +164,29 @@ void update()
 					e.getValue().setLocation("<html>" + locs.stream().collect(Collectors.joining("<br>")) + "</html>");
 				}
 			}
+
+			final ItemContainer itemContainer = client.getItemContainer(InventoryID.INVENTORY);
```

Comment:
This should be run on the client thread. Also, the plugin already keeps track of books the player is holding--this doesn't need to be calculated again.